pkg/transforms/genericresource.go

```diff
@@ -4,32 +4,49 @@
 package transforms
 
 import (
+	"fmt"
 	"strings"
 	"time"
 
 	"github.com/stolostron/search-collector/pkg/config"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/util/jsonpath"
+	"k8s.io/klog/v2"
 )
 
 // GenericResource ...
 type GenericResource struct {
 	node Node
 }
 
-// GenericResourceBuilder ...
-// Builds a GenericResource node. Extract the useful properties from unstructured resource.
+// Builds a GenericResource node.
+// Extract default properties from unstructured resource.
+// Supports extracting additional properties defined by the transform config.
 func GenericResourceBuilder(r *unstructured.Unstructured) *GenericResource {
 	n := Node{
 		UID:        prefixedUID(r.GetUID()),
-		Properties: unstructuredProperties(r),
-		Metadata:   make(map[string]string),
+		Properties: genericProperties(r),
+		Metadata:   genericMetadata(r),
 	}
-	n.Metadata["OwnerUID"] = ownerRefUID(r.GetOwnerReferences())
-	//Adding OwnerReleaseName and Namespace for the resources that doesn't have ownerRef, but are deployed by a release
-	if n.Metadata["OwnerUID"] == "" && r.GetAnnotations()["meta.helm.sh/release-name"] != "" &&
-		r.GetAnnotations()["meta.helm.sh/release-namespace"] != "" {
-		n.Metadata["OwnerReleaseName"] = r.GetAnnotations()["meta.helm.sh/release-name"]
-		n.Metadata["OwnerReleaseNamespace"] = r.GetAnnotations()["meta.helm.sh/release-namespace"]
+
+	// Check if a transform config exists for this resource and extract the additional properties.
+	transformConfig, found := getTransformConfig(r.GroupVersionKind().Group, r.GetKind())
+	if found {
+		for _, prop := range transformConfig.properties {
+			jp := jsonpath.New(prop.name)
+			parseErr := jp.Parse(prop.jsonpath)
+			if parseErr != nil {
+				klog.Errorf("Error parsing jsonpath %s for property %s: %v", prop.jsonpath, prop.name, parseErr)
```

Comment:
Will including the kind help?
```suggestion
				klog.Errorf("Error parsing jsonpath %s for kind %s and property %s: %v", prop.jsonpath, r.GetKind(), prop.name, parseErr)
```

Reply:
Yes, kind and group needed to debug errors in the config.  Updated all error messages.